python/looker/rtl/user_session.py

```diff
@@ -61,48 +52,58 @@ def get_token(self):
             self.login()
         return self._token
 
-    def authenticate(self):
-         # TODO: still to be built based on request
-        return
+    def authenticate(self) -> Dict[str, str]:
+        token = self.get_token()
+        header = {}
+        if token and token.access_token:
+            header['Authorization'] = f'token {token.access_token}'
+        return header
 
     def login(self, user_id: Optional[str] = None) -> at.AuthToken:
         if not self.is_authenticated:
-            token = self.__login(user_id)
+            token = self._login(user_id)
             self._token = at.AuthToken(token)
-            return token
         return self._token
 
-    def __login(self, user_id: Optional[str]) -> tp.SDKSuccessResponse.value:
+    def _login(self, user_id: Optional[str]) -> tp.Response.value:
         if (user_id and self.user_id != user_id):
             # We are switching user ids
-            self.__logout()
+            self._logout()
 
-        self.__reset()
+        self._reset()
 
         # pylint: disable=too-many-function-args
-        result = self.__ok(self.transport.request(tp.HttpMethod.POST,
-                                                  '/login',
-                                                  body={
-                                                      'client_id': self.settings.client_id,
-                                                      'client_secret': self.settings.client_secret
-                                                  }))
+        result = self._ok(
+            self.transport.request(tp.HttpMethod.POST,
+                                   '/login',
+                                   body={
+                                       'client_id': self.settings.client_id,
+                                       'client_secret':
+                                       self.settings.client_secret
+                                   }))
         return result
 
     def logout(self) -> bool:
-        result = False
         if self.is_authenticated:
-            result = self.__logout()
-        return result
+            result = self._logout()
+        return bool(result)
```

Comment:
`result` won't exist if the `self.is_authenticated` condition is false. I think you should keep the removed L92. I wonder what meaning the return value of `logout()` should have ... `True` - "you were logged in and now you're logged out", `False` - "you weren't logged in dummy" | "we failed to log you out" Might be simplest for now to leave the return value as `None` and raise an exception if the API or network barfed.

Reply:
I think I intended logout `false` to mean "no logout happened, probably because you weren't logged in." Maybe return `true` either way, meaning "yeah, you're now definitely logged out?"